### PR TITLE
Add blog post to summarize upcoming spec changes

### PR DESCRIPTION
**Pull request by jdolitsky**

### CODE REVIEW

File: content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+of the individual specifications are all technically independent of each other.
+
+However, when dealing with clients responsible with pushing/pulling from
+registries, or registries themselves, hearing claims of "support for OCI 1.1"
+can be assumed to be relating to reference types and the changes
+outlined in this post.
+
+# List of Major Changes (3)
+
+## 1. New Registry API Endpoint for Querying Relationships
```
Comment (tianon) [inline content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md:34]: IMO, we should swap this ordering:
1. guidance for generic artifacts
2. means of specifying relationships between objects (images, artifacts, and indexes alike)
3. API for querying those relationships

I realize the API is what a lot of folks are here for, but I think it's important to set the context for why the new API exists and what it's providing before we introduce it (since the target of this post is probably folks who _haven't_ been following along, not those of us who are now too familiar).

File: content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+}
+```
+
+This is the primary mechanism for linking objects to one another. This
+is what is used to determine the proper list returned by the API endpoint
+described in #1 above.
+
+If the registry supports the processing of the `subject` field, it is required
+to respond with a header in the following form: `OCI-Subject: sha256:5b0bca...`
+( the value being the `digest` field in the `subject` JSON object).
```
Comment (tianon) [inline content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md:108]:
```suggestion
(the value being the `digest` field in the `subject` JSON object).
```

**Review by tianon — COMMENTED** [review]
:bow: :heart: thanks for working on this!

**Comment by sudo-bmitch** [comment]
Other things I can think of to add:

- `data` field in the descriptor
- non-distributable layers are deprecated
- zstd compression support
- extension support in distribution-spec
- support for resuming chunked blob push
- anonymous blob mount support in distribution-spec

File: content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md (hunk 1 above)
Comment (jdolitsky) [reply content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md:34]: I have no problem switching the ordering. Are you suggesting those as new headers too? Or do the current ones work?

**Comment by jdolitsky** [comment]
@sudo-bmitch - in the final section "Closing Thoughts" I mention that this is just the major stuff. How do you feel about just adding your list as bullets at the end, but not going into detail?

File: content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md (hunk 1 above)
Comment (tianon) [reply content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md:34]: The current ones don't bother me too much, just the ordering (as noted by the fact that you then have to say "refer back to item 1" in both the second and third section). :sweat_smile:

If I wanted to nitpick I might reword "Official Guidance on How to Specify Alternatative Artifacts" as something like "Official Guidance on How to Create and Store Alternative (Even Non-Container) Artifacts" (including the typo fix for "Alternatative" that I didn't notice until typing this comment :smile:), but it's getting a bit wordy and perhaps should be shorter. :see_no_evil: :heart:

**Comment by sudo-bmitch** [comment]
Oh I was just keeping it to the big stuff, and I'm not even sure how to word that registries cannot implement an allow list of artifacts in this, which also feels big to me. Or the change in the regexp for the repository value. But keeping the description to a short bulleted list works.

File: content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md (hunk 1 above)
Comment (jdolitsky) [reply content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md]: done!

**Comment by jdolitsky** [comment]
@sudo-bmitch @tianon - just added some updates to address your feedback, please lmk how that looks

File: content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+Some other notable changes include the following:
+
+- `data` field in the descriptor
+- non-distributable layers are deprecated
+- zstd compression support
+- extension support in distribution-spec
+- support for resuming chunked blob push
+- anonymous blob mount support in distribution-spec
+
```
Comment (tianon) [inline content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md:152]: https://github.com/opencontainers/distribution-spec/pull/393 might be a good one to add here ("a `Warning` header for surfacing user-facing client warnings" or something) :eyes:

Comment (jdolitsky) [reply content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md:153]: good call - done

**Review by sudo-bmitch — APPROVED** [review]
LGTM

File: content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+in the OCI Image and Distribution specs to enable the creation and
+discovery of relationships between objects stored in an OCI registry.
+
+These changes (with some subsequent modifications) will be included in
+upcoming minor releases of both OCI Image and Distribution spec.
+
+# Notes on Versioning and the Term "OCI 1.1"
+
+The version of the upcoming minor releases of both OCI Image and
+Distribution spec happens to be the same: `v1.1.0`.
```
Comment (h-vetinari) [inline content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md:21]: It seems opencontainers-specs are self-aligning, as the runtime spec is also looking to release [1.1.0](https://github.com/opencontainers/runtime-spec/pull/1213) any moment now...

Comment (jdolitsky) [reply content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md:21]: yes, but not intentional 😅 may not always be the case in the future
